smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/ContentTypeMiddleware.kt

```diff
@@ -0,0 +1,35 @@
+package software.amazon.smithy.swift.codegen.integration.middlewares
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.middleware.MiddlewarePosition
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderable
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareStep
+import software.amazon.smithy.swift.codegen.model.capitalizedName
+
+class ContentTypeMiddleware(val defaultContentType: String) : MiddlewareRenderable {
+
+    override val name = "ContentTypeMiddleware"
+
+    override val middlewareStep = MiddlewareStep.SERIALIZESTEP
+
+    override val position = MiddlewarePosition.AFTER
+
+    override fun render(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+        serviceShape: ServiceShape,
+        op: OperationShape,
+        operationStackName: String
+    ) {
+        val inputShape = ctx.model.expectShape(op.input.get())
+        val inputShapeName = ctx.symbolProvider.toSymbol(inputShape).name
+        val outputShape = ctx.model.expectShape(op.output.get())
+        val outputShapeName = ctx.symbolProvider.toSymbol(outputShape).name
+        val outputErrorName = "${op.capitalizedName()}OutputError"
```

Comment:
same thing here, can we use the static functions inside `ServiceGenerator` to eliminate duplicated code here?